tests/test_dense.py

```diff
@@ -0,0 +1,69 @@
+from qutip_cupy import CuPyDense
+import numpy as np
+import pytest
+
+@pytest.fixture(scope="function", params=((1, 2), (5, 10), (7, 3), (2, 5)))
+def shape(request):
+    return request.param
+
+
+class TestCuPyDenseDispatch:
+    """ Tests if the methods and conversions have been
+        succesfully registered to QuTiP's Data Layer."""
+
+    def test_conversion_cycle(self, shape):
+
+        from qutip.core import data
+
+        qutip_dense = data.Dense(np.random.uniform(size=shape))
+
+        tr1 = data.to(CuPyDense, qutip_dense)
+        tr2 = data.to(data.Dense, tr1)
+
+        np.testing.assert_array_equal(qutip_dense.to_array(), tr2.to_array())
+
+class TestCuPyDense:
+    """ Tests of the methods and constructors of the CuPyDense class. """
+
+    def test_shape(self, shape):
+
+        cupy_dense = CuPyDense(np.random.uniform(size=shape))
+
+        assert (cupy_dense.shape == shape)
```

Comment:
```suggestion
        cupy_dense = CuPyDense(np.random.uniform(size=shape))
        assert (cupy_dense.shape == shape)
```
No need to have blank lines in short functions -- they just make the file longer and harder to read, I think.